Sources/Prometheus/MetricTypes/Gauge.swift

```diff
@@ -1,9 +1,11 @@
-/// Prometheus Counter metric
+import NIOConcurrencyHelpers
+
+/// Prometheus Gauge metric
 ///
 /// See https://prometheus.io/docs/concepts/metric_types/#gauge
-public class Gauge<NumType: Numeric, Labels: MetricLabels>: Metric, PrometheusHandled {
+public class PromGauge<NumType: Numeric, Labels: MetricLabels>: Metric, PrometheusHandled {
     /// Prometheus instance that created this Gauge
-    internal let prometheus: PrometheusClient
+    internal weak var prometheus: PrometheusClient?
```

Comment:
Not super sure they really have to be `weak` to be honest... are we sure about that?
In an app lifecycle you'd generally boostrap the metrics into the `MetricsSystem` and then it's there "forever"; what is the weak gaining us?

Reply:
+1

Reply:
I might be off on my cycles here, but it's not about not being able to drop the PrometheusClient, but about not being able to drop the Gauge if it holds on to it's client, and the client holds on to the Gauge (it's in it's `metrics` array)